app/helpers/fhir_helper.rb

```diff
@@ -1032,51 +1055,49 @@ def from_exposure_risk_factors_extension(patient)
           }
       when "#{SA_EXT_BASE_URL}laboratory-personnel"
         sub_ext_risk_factors =
           {
-            laboratory_personnel: from_bool_extension_false_default(sub_ext, "Patient.extension[#{ext_idx}].extension[#{sub_ext_idx}]", 'laboratory-personnel'),
-            laboratory_personnel_facility_name: from_string_extension(sub_ext, "Patient.extension[#{ext_idx}].extension[#{sub_ext_idx}]",
-                                                                      'laboratory-personnel-facility-name')
+            laboratory_personnel: from_bool_extension_false_default(sub_ext, base_path, 'laboratory-personnel'),
+            laboratory_personnel_facility_name: from_string_extension(sub_ext, base_path, 'laboratory-personnel-facility-name')
           }
       when "#{SA_EXT_BASE_URL}healthcare-personnel"
         sub_ext_risk_factors =
           {
-            healthcare_personnel: from_bool_extension_false_default(sub_ext, "Patient.extension[#{ext_idx}].extension[#{sub_ext_idx}]", 'healthcare-personnel'),
-            healthcare_personnel_facility_name: from_string_extension(sub_ext, "Patient.extension[#{ext_idx}].extension[#{sub_ext_idx}]",
-                                                                      'healthcare-personnel-facility-name')
+            healthcare_personnel: from_bool_extension_false_default(sub_ext, base_path, 'healthcare-personnel'),
+            healthcare_personnel_facility_name: from_string_extension(sub_ext, base_path, 'healthcare-personnel-facility-name')
           }
       when "#{SA_EXT_BASE_URL}member-of-a-common-exposure-cohort"
-        sub_ext_risk_factors =
-          {
-            member_of_a_common_exposure_cohort: from_bool_extension_false_default(sub_ext, "Patient.extension[#{ext_idx}].extension[#{sub_ext_idx}]",
-                                                                                  'member-of-a-common-exposure-cohort'),
-            member_of_a_common_exposure_cohort_type: from_string_extension(sub_ext, "Patient.extension[#{ext_idx}].extension[#{sub_ext_idx}]",
-                                                                           'member-of-a-common-exposure-cohort-type')
-          }
-      else
-        next
+        common_exposure_cohort = {}
+        %i[cohort_type cohort_name cohort_location].each do |cohort_field|
+          cohort_field_value = from_string_extension(sub_ext, base_path, "member-of-a-common-exposure-#{cohort_field.to_s.dasherize}")
+          common_exposure_cohort[cohort_field] = cohort_field_value[:value] if cohort_field_value[:value].present?
+        end
+        common_exposure_cohorts << common_exposure_cohort
       end
+
+      next if sub_ext_risk_factors.nil?
+
       risk_factors.merge!(sub_ext_risk_factors.transform_values { |v| { value: v[:value], path: v[:path].sub(SA_EXT_BASE_URL, '') } })
     end
+    risk_factors[:common_exposure_cohorts_attributes] = { value: common_exposure_cohorts } if common_exposure_cohorts.present?
```

Comment:
This implementation ends up being additive in updates. Aka if I have a patient with 6 cohorts, and I try to update just one of the cohorts, this will create 6 new cohorts, instead of just updating the one cohort. That isn't how we want our implementation of update to work, so to really do this right, I think we truly need to just entirely recreate the cohorts upon update. This seems inefficient, but I can't think of a better way.

Reply:
good catch: https://github.com/SaraAlert/SaraAlert/pull/1125/commits/237545888b608d51912da918ac3e3e6d0780ee28#diff-402d451d1f7019322100522f6898c3153444edb8ddb963e181a457c84a2ca6b6R1102